cmd/gardener-apiserver/app/gardener_apiserver.go

```diff
@@ -111,19 +105,18 @@ type Options struct {
 func NewOptions(out, errOut io.Writer) *Options {
 	o := &Options{
 		Recommended: genericoptions.NewRecommendedOptions(
-			fmt.Sprintf("/registry/%s", garden.GroupName),
+			"/registry/gardener.cloud",
```

Comment:
:warning: This will break existing Gardener installations.

Today, we store all Gardener-related resources with the `/registry/garden.sapcloud.io` prefix.
Also, resources created with the old API group look like this in etcd:
* `/registry/garden.sapcloud.io/garden.sapcloud.io/shoots/namespaces/foo/bar`

We would like to migrate to `/registry/gardener.cloud` prefix such that resources look in etcd like this:
* `/registry/gardener.cloud/core.gardener.cloud/shoots/namespaces/foo/bar`
* `/registry/gardener.cloud/settings.gardener.cloud/whatever/....`

Together with @mvladev we agreed on a special migration action that we require users to perform when upgrading to the `v1.0` Gardener version (we bump the major version!).
The idea is to

1. Stop all apiservers
1. Do a full-snapshot backup of ETCD
1. Get all keys with `/registry/garden.sapcloud.io` prefix
1. Put them with new `/registry/gardener.cloud` prefix and rename `garden.sapcloud.io` to `core.gardener.cloud`
1. Scale up all apiservers
1. Let them run
1. Delete all resources with old `/registry/garden.sapcloud.io` prefix

We are working on a small go program based on https://github.com/etcd-io/etcd/tree/master/clientv3 that will do this job (at least steps 2, 3, 4, and 7). Once ready, we will write a small migration guide and remove the "draft" status from this PR. 

Though, reviews are already very welcome now, so please take a look already.